src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "../libraries/BN254.sol";
+import "forge-std/Test.sol";
+
+
+contract IndexRegistry is IIndexRegistry, Test {
+
+    IRegistryCoordinator public registryCoordinator;
+
+    // list of all unique registered operators
+    bytes32[] public globalOperatorList;
+
+    // mapping of operatorId => quorumNumber => index history of that operator
+    mapping(bytes32 => mapping(uint8 => OperatorIndex[])) public operatorIdToIndexHistory;
+    // mapping of quorumNumber => history of numbers of unique registered operators
+    mapping(uint8 => OperatorIndex[]) public totalOperatorsHistory;
+
+    modifier onlyRegistryCoordinator() {
+        require(msg.sender == address(registryCoordinator), "IndexRegistry.onlyRegistryCoordinator: caller is not the registry coordinator");
+        _;
+    }
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator
+    ){
+        registryCoordinator = _registryCoordinator;
+    }
```

Comment:
maybe even combine this + the storage into a `CoordinatedRegistry` or `RegistryWorker` or something that other registries can inherit from?
probably overkill, just expressing an idea that popped into my head, feel free to resolve this + ignore.

Reply:
i think it's a fair suggestion. rejecting it
- out of laziness
- more contracts
- more to pay attention to when upgrading
- not too much abstracted code